api/v2beta1/ratelimitpolicy_types.go

```diff
@@ -0,0 +1,218 @@
+limitations under the License.
+*/
+
+package v2beta1
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/go-logr/logr"
+	"github.com/google/go-cmp/cmp"
+	"github.com/kuadrant/kuadrant-operator/pkg/common"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gatewayapiv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ContextSelector defines one item from the well known selectors
+// TODO Document properly "Well-known selector" https://github.com/Kuadrant/architecture/blob/main/rfcs/0001-rlp-v2.md#well-known-selectors
+// +kubebuilder:validation:MinLength=1
+// +kubebuilder:validation:MaxLength=253
+type ContextSelector string
+
+// +kubebuilder:validation:Enum:=eq;neq
+type WhenConditionOperator string
+
+// +kubebuilder:validation:Enum:=second;minute;hour;day
+type TimeUnit string
+
+// Rate defines the actual rate limit that will be used when there is a match
+type Rate struct {
+	// Limit defines the max value allowed for a given period of time
+	Limit int `json:"limit"`
+
+	// Duration defines the time period for which the Limit specified above applies.
+	Duration int `json:"duration"`
+
+	// Duration defines the time uni
+	// Possible values are: "second", "minute", "hour", "day"
+	Unit TimeUnit `json:"unit"`
+}
+
+// RouteSelector defines semantics for matching an HTTP request based on conditions
+// https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteSpec
+type WhenCondition struct {
+	// Selector defines one item from the well known selectors
+	// TODO Document properly "Well-known selector" https://github.com/Kuadrant/architecture/blob/main/rfcs/0001-rlp-v2.md#well-known-selectors
+	Selector ContextSelector `json:"selector"`
+
+	// The binary operator to be applied to the content fetched from the selector
+	// Possible values are: "eq" (equal to), "neq" (not equal to), "incl" (includes; for arrays), "excl" (excludes; for arrays), "matches" (regex)
+	Operator WhenConditionOperator `json:"operator"`
+
+	// The value of reference for the comparison.
+	// If used with the "matches" operator, the value must compile to a valid Golang regex.
+	Value string `json:"value"`
+}
+
+// RouteSelector defines semantics for matching an HTTP request based on conditions
+// https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteSpec
+type RouteSelector struct {
+	// Hostnames defines a set of hostname that should match against the HTTP Host header to select a HTTPRoute to process the request
+	// https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteSpec
+	// +optional
+	Hostnames []gatewayapiv1alpha2.Hostname `json:"hostnames,omitempty"`
```

Comment:
This means a validation (which I like!) against the following pattern:
```
^(\*\.)?[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$
```

Only to highlight that the example provided in the description of the PR (`*example.com`) is **not** valid.

Reply:
good catch! I will change it.